activemodel/lib/active_model/attribute_methods.rb

```diff
@@ -365,6 +419,13 @@ def generated_attribute_methods
           @generated_attribute_methods ||= Module.new.tap { |mod| include mod }
         end
 
+        def generated_alias_attribute_methods
+          @generated_alias_attribute_methods ||= Module.new.tap do |mod|
```

Comment:
I don't understand why we do this in a different module. Can't we generated these methods in `@generated_attribute_methods`?

Reply:
We should be able to generate it on `@generated_attribute_methods`. Unless I forgot about some failures that were preventing us from doing that, the only reason was that I didn't want to touch `@generated_attribute_methods` by adding mutex to it since Active Model's `@generated_attribute_methods` doesn't have it

I'm going to try using the same module for both aliases and attributes

Reply:
I remembered why I chose to have a separate module. Currently `undefine_attribute_methods` completely wipes `generated_attribute_methods` module which means it will wipe alias attributes as well. Since currently alias attribute methods and attribute methods are "detached"  from each other meaning that `undefine_attribute_methods` won't affect alias attribute methods I decided to keep them separate. 

But I think I should be able to tweak `undefine_attribute_methods` to make sure it resets flags that control alias attribute methods as well

Reply:
> Since currently alias attribute methods and attribute methods are "detached" from each other meaning that

Can we make them not detached? Ideally they should all be defined together.

If we properly do that, it means we'll be able to define aliases with actual `alias_method` rather than to create these small delegators, the pros:

  - Less memory usage
  - Some compilation unit for YJIT
  - Behaves more like `alias_method`, which users likely expect.

Reply:
Done, now we will generate both attribute methods and alias attribute methods on the same `generated_attribute_methods` module. It was easy to achieve, I just made sure `undefine_attribute_methods` resets state of the alias attribute generation logic